public/js/widget-manager.js

```diff
@@ -45,16 +61,73 @@ define(['jupyter-js-widgets'], function(Widgets) {
         });
     };
 
+    /*
+     * Returns callbacks to be invoked in response to ipywidget triggered
+     * comm messages on shell, input, and iopub channels. Each callback takes
+     * one parameter: the comm message from the kernel. The return value should
+     * be an object with any of the following properties defined as functions.
+     *
+     * iopub.status
+     * iopub.clear_output
+     * iopub.output
+     * shell.reply
+     * input
+     *
+     * view: ipywidget view instance
+     */
+    WidgetManager.prototype.callbacks = function(view) {
+        // TODO: only registering one callback until https://github.com/ipython/ipywidgets/pull/353
```

Comment:
This needs to change once the bug is fixed in ipywidgets. But the callback API might change in the shim too.

Since we pin to a git sha for ipywidgets today, this "fix" will continue to work until we move on.